src/Introspection.cpp

```diff
@@ -2200,7 +2201,13 @@ class DebugSections {
 };
 
 namespace {
+
 DebugSections *debug_sections = nullptr;
+
+#if __has_feature(thread_sanitizer)
```

Comment:
I'm guessing this won't work on MSVC. It seems fine to just put the mutex in unconditionally, the performance of this doesn't matter does it?

Reply:
You're right, needs to be wrapped in #if defined(__has_feature), as it is in Target.cpp.

It's called from `make_entity_name()`, which is called a lot.

Reply:
I think if you just put the lock after the two early outs for missing debug data, it will only ever actually lock the mutex for make_entity_name calls directly from user declared objects. I think all of the internal make_entity_name calls won't get that far.

Reply:
I think they all do get that far. The early-outs are for when the user code wasn't compiled with -g, so the stack walking also occurs for things internal to libHalide.